src/cli/sandbox.cpp

```diff
@@ -63,6 +67,56 @@ bool Sandbox::init()
       }
 
    return (::cap_enter() == 0);
+#elif defined(BOTAN_TARGET_OS_HAS_SETPPRIV)
+   priv_set_t *ps;
+
+   if ((ps = ::priv_allocset()) == nullptr)
```

Comment:
Can we use `unique_ptr<priv_set_t, ::priv_freeset>` to avoid the code duplication of freeing on error?